core/types/celo_transaction.go

```diff
@@ -0,0 +1,41 @@
+package types
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/exchange"
+)
+
+// CompareWithRates compares the effective gas price of two transactions according to the exchange rates and
+// the base fees in the transactions currencies.
+func CompareWithRates(a, b *Transaction, goldBaseFee *big.Int, ratesAndFees *exchange.RatesAndFees) int {
+	if ratesAndFees == nil {
+		// During node startup the ratesAndFees might not be yet setup, compare nominally
+		feeCapCmp := a.GasFeeCapCmp(b)
+		if feeCapCmp != 0 {
+			return feeCapCmp
+		}
+		return a.GasTipCapCmp(b)
+	}
+	rates := ratesAndFees.Rates
+	if goldBaseFee != nil {
+		tipA := a.EffectiveGasTipValue(ratesAndFees.GetBaseFeeIn(a.inner.feeCurrency()))
+		tipB := b.EffectiveGasTipValue(ratesAndFees.GetBaseFeeIn(b.inner.feeCurrency()))
+		result, _ := exchange.CompareValue(rates, tipA, a.inner.feeCurrency(), tipB, b.inner.feeCurrency())
+		return result
+	}
+
+	// Compare fee caps if baseFee is not specified or effective tips are equal
```

Comment:
I just noticed that even if effective tips above are equal we won't get to this code anyway.